src/libutil/error.cc

```diff
@@ -200,13 +200,125 @@ std::ostream & showErrorInfo(std::ostream & out, const ErrorInfo & einfo, bool s
 
     auto noSource = ANSI_ITALIC " (source not available)" ANSI_NORMAL "\n";
 
-    // traces
-    if (showTrace && !einfo.traces.empty()) {
+    /*
+     * Traces
+     * ------
+     *
+     *  The semantics of traces is a bit weird. We have only one option to
```

Comment:
It depends where you stand from I think. Its the semantics of traces that is weird in a lazy & declarative programming language with respect to imperative traces. They do not mean the same thing.
But yes, the implementation of traces here is weird, just because we cannot do it differently given the nix language features.